pkg/authorization/rewrite/rewrite_test.go

```diff
@@ -224,6 +233,7 @@ func createRequest(queryParams, headers map[string][]string) *http.Request {
 			}
 		}
 		r.URL.RawQuery = q.Encode()
+		r.URL, _ = url.Parse(r.URL.String())
```

Comment:
To avoid mutating the original `*url.URL`, it would make sense to copy it above the `r.URL.RawQuery = q.Encdoe()`.